### DISCUSSION TITLE
IntelliJ: Package name prefix

### DISCUSSION THREAD

**Discussion by tcf909** [invertase/melos#336] [Q&A] [answered]
Anyway to eliminate the package name prefix that Melos seems to use for intellij packages?

We already prefix with a lot of things, one more thing that doesn't match the directory structure is messy.

/root/tftc_flutter_app_name

ends up being melos_tftc_flutter_app_name

**Comment by ryanhanks** [comment]
Just came here to ask this.

Looks like it's hard-coded: https://github.com/invertase/melos/blob/main/packages/melos/lib/src/common/intellij_project.dart#L107-L118

I'm wondering if there's a reason to namespace like this. In terms of my workflow I have no conflicts if the default package names had no 'melos_' prefix, but if it's needed, let's figure out a way to at least override this.

Looks like @Salakar and @blaugold have been around here recently, do either of you have input on this? I'm happy to put up a PR, just need some background to proceed.

**Reply by Salakar** [reply]
The main reason for the prefix was just a way to avoid editing/overriding users own configuration files, since it has the prefix and matched a certain naming convention we know we can safely make changes.

Open to PRs, could put it behind a bootstrap option in melos.yaml or something like that

**Reply by ryanhanks** [reply]
Thanks Salakar.

I think it makes sense to try to continue to use the "ide/intellij" object in melos.yaml for all the intelliJ config. As pointed out by [this comment](https://github.com/invertase/melos/blob/main/packages/melos/lib/src/workspace_configs.dart#L80), the current config parser only supports boolean values for "ide/intellij". I took [a first pass](https://github.com/invertase/melos/compare/main...ryanhanks:melos:add_support_for_intellij_module_name_prefix?expand=1) at supporting that while also allowing "ide/intellij" to be a map, i.e.,
```yaml
ide:
  intellij:
    enabled: true
```
The branch currently has the config parser looking for a module name prefix value in the "moduleNamePrefix" key:
```yaml
ide:
  intellij:
    moduleNamePrefix: ''  # set this to empty to override the default
```
Any thoughts or ideas on this approach?

**Reply by blaugold** [reply]
👍 That's what I would have done.

**Reply by ryanhanks** [reply]
Thanks blaugold.

I'll put up a PR shortly

**Reply by ryanhanks** [reply]
Added some comments to the PR: https://github.com/invertase/melos/pull/349

Take a look when you get a chance and lmk what you think

**Comment by ryanhanks-bestow** [comment, marked as answer]
@tcf909 the IntelliJ package name prefix can now be configured, and set to the empty string to get the outcome you're after. See https://melos.invertase.dev/configuration/overview#ideintellijmodulenameprefix
